src/main/java/io/vertx/core/spi/tracing/VertxTracer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2011-2017 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.core.spi.tracing;
+
+import io.vertx.core.Context;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public interface VertxTracer<I, O> {
```

Comment:
I don't think usually client and server policy is specified in the same interface as the library impls usually differ. I would recommend splitting these out as client and server.

Also, notice how even outside brave, the data policy opinions are not on the callback interface. Here's another example https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/ClientStreamTracer.java

Reply:
do you mean this interface should be split in 2 interfaces ?